### DISCUSSION TITLE
SS能否內建DNS快取功能？

### DISCUSSION THREAD

**Discussion by okudayukiko** [shadowsocks/shadowsocks-rust#515] [Ideas]
SS能否內建DNS快取功能？類似dnsmasq。
以英國WireGuard為例，如果用1.1.1.1 DNS，就會時快時慢，可以看YouTube HD，但是使用Facebook、Twitter、Niconico就很慢；使用英國DNS，時快時慢問題解決。

**Comment by zonyitoo** [comment]
shadowsocks is a level-4 proxy (TCP and UDP). It doesn't provide any services related to DNS resolving.

I personally suggest https://github.com/pymumu/smartdns .

**Reply by okudayukiko** [reply]
V2Ray已經內建DNS快取功能。

**Reply by zonyitoo** [reply]
v2ray is a set of programs, but shadowsocks is only a level-4 proxy protocol.

**Reply by database64128** [reply]
> V2Ray已經內建DNS快取功能。

The DNS cache in V2Ray is only for the router. V2Ray doesn’t provide DNS service.

**Reply by okudayukiko** [reply]
https://www.v2fly.org/config/dns.html
V2Ray預設開啟DNS Cache

**Reply by database64128** [reply]
V2Ray doesn't provide a DNS server for you. If you send a DNS query via V2Ray, it's not cached. The DNS cache is only utilized by the router to match IP rules if you use domain strategies that involve the resolution of domain names.

**Comment by om2mo** [comment]
SS本身就不提供解析，何来快取。它和V2RAY一样发送给相关的DNS。
